modules/clipboard.nix

```diff
@@ -0,0 +1,63 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+} @ args:
+with lib; let
+  helpers = import ../lib/helpers.nix args;
+  cfg = config.clipboard;
+in {
+  options = {
+    clipboard = {
+      register = mkOption {
+        description = ''
+          Sets the register to use for the clipboard.
+          Learn more at https://neovim.io/doc/user/options.html#'clipboard'.
+        '';
+        type = with types;
+          nullOr
+          (either str (listOf str));
+        default = null;
+        example = "unnamedplus";
+      };
+
+      provider = mkOption {
+        description = ''
+          Package to use as the clipboard provider.
+          Learn more at `:h clipboard`.
+        '';
+        type = with types;
+          nullOr (
+            either package
+            (enum ["xclip" "xsel"])
+          );
+        default = null;
+        example = "xclip";
+      };
+    };
+  };
+
+  config = {
+    extraConfigLuaPre =
```

Comment:
Can't we use the nixvim options.clipboard ?

Reply:
Definitely. Done